core/src/main/java/com/novoda/merlin/service/MerlinService.java

```diff
@@ -67,10 +68,15 @@ private void disableConnectivityReceiver() {
 
     private void setReceiverState(int receiverState) {
         if (USE_COMPONENT_ENABLED_SETTING) {
-            getPackageManager().setComponentEnabledSetting(new ComponentName(this, ConnectivityReceiver.class), receiverState, PackageManager.DONT_KILL_APP);
+            getPackageManager().setComponentEnabledSetting(connectivityReceiverComponent(), receiverState, PackageManager.DONT_KILL_APP);
         }
     }
 
+    @VisibleForTesting
+    protected ComponentName connectivityReceiverComponent() {
```

Comment:
I don't like this, but it's not like we can create services ourselves, passing in a ComponentName factory or something. (Same with the ComponentReceiver in the other PR.)